frontend-react/src/components/FileHandlers/FileHandler.tsx

```diff
@@ -250,6 +251,19 @@ const FileHandler = ({
         (reportId || overallStatus === OverallStatus.VALID) &&
         !hasQualityFilterMessages;
 
+    // Once file is submitted, send to analytics
+    if (submitted && !isWorking) {
```

Comment:
Can we move this block to after [the `sendFile` above](https://github.com/CDCgov/prime-reportstream/pull/8057/files#diff-90daad087bcd5ce4dac7318290253991cf7355a69bdc37793105cdb8b08e7ae1R182)?  It'd ensure we'd only track the event once rather than on every render when `submitted && !isWorking`.

Reply:
@stephenkao  It would need to be after submitted. The submitted is not declared until after the sendFile. Options?

Reply:
`sendFile` from useWatersUploader [actually submits it to the `/validate` endpoint](https://github.com/CDCgov/prime-reportstream/blob/master/frontend-react/src/hooks/network/WatersHooks.ts#L25-L52), right?  `submitted` there is just a derived value that happens after a re-render, but we'll have the response from the `async sendFile` call:

```js
const response =  await sendFile(...);

// can access response.errorCount, response.warningCount, et cetera
```

This may actually need to be wrapped in a `try-catch` since I'm pretty sure invalid validations come back as an error code rather than a 200, though!

Reply:
@stephenkao good catch!